Source/ApiTemplate/Source/ApiTemplate/MvcBuilderExtensions.cs

```diff
@@ -0,0 +1,135 @@
+namespace ApiTemplate
+{
+    using System.Linq;
+    using System.Text.Json;
+    using System.Text.Json.Serialization;
+    using ApiTemplate.Options;
+    using Boxed.AspNetCore;
+    using Microsoft.AspNetCore.Hosting;
+    using Microsoft.AspNetCore.JsonPatch;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.AspNetCore.Mvc.Formatters;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Hosting;
+    using Microsoft.Extensions.Options;
+
+    public static class MvcBuilderExtensions
+    {
+        /// <summary>
+        /// Adds customized JSON serializer settings.
+        /// </summary>
+        public static IMvcBuilder AddCustomJsonOptions(
+            this IMvcBuilder builder,
+            IWebHostEnvironment webHostEnvironment) =>
+            builder.AddJsonOptions(
+                options =>
+                {
+                    var jsonSerializerOptions = options.JsonSerializerOptions;
+                    if (webHostEnvironment.IsDevelopment())
+                    {
+                        // Pretty print the JSON in development for easier debugging.
+                        jsonSerializerOptions.WriteIndented = true;
+                    }
+
+                    jsonSerializerOptions.Converters.Add(new JsonStringEnumConverter());
+                    jsonSerializerOptions.DictionaryKeyPolicy = JsonNamingPolicy.CamelCase;
+                    jsonSerializerOptions.PropertyNamingPolicy = JsonNamingPolicy.CamelCase;
+                });
+
+        public static IMvcBuilder AddCustomMvcOptions(this IMvcBuilder builder) =>
+            builder.AddMvcOptions(
+                options =>
+                {
+                    // Controls how controller actions cache content from the appsettings.json file.
+                    var cacheProfileOptions = builder
+                        .Services
+                        .BuildServiceProvider()
```

Comment:
I don't think you want to do this on startup. This will create a second instance of all singletons registered within the app. Why not just get the `CacheProfileOptions` from Configuration again and bind with `.Get<CacheProfileOptions>`?

It might be okay to do this if we don't do anything crazy when any of our singleton instances are created.

https://github.com/aspnet/AspNetCore/issues/14587#issuecomment-553445007

Reply:
This might be a good idea. I'm on holiday and having trouble finding time to do the last 5% and release. I'll see if I can take a look today.

Reply:
I was wondering where you vanished to ;)